### PR TITLE
[github-actions] Made it so tests also run on intel

### PR DESCRIPTION
**Pull request by danbugs**
Note: I only modified the dep_rust matrix which will run the tests to assess differences between amd and intel. In the future, we might want to consider revisiting our matrix setup.

### CODE REVIEW

**Comment by ludfjig** [comment]
Could we add intel to benchmarking too? And include them in github release artifacts?

**Comment by simongdavies** [comment]
@danbugs when this merges you will need to update the branch protection rules required checks as there are now new checks and some of the existing ones don't exist

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
-<div align="center">
+<div style="display: grid; place-items: center;">
```
Comment (ludfjig) [inline README.md]: Did you intend to include these changes in this PR? I think it would be better to make a separate PR for this

File: .github/workflows/Benchmarks.yml [hunk 2]
```diff
@@ -17,17 +17,39 @@ jobs:
...
+          linux-hyperv-release-amd,
+          windows-2022-release-intel,
+          linux-kvm-release-intel,
+          linux-hyperv-release-intel,
+        ]
         include:
-          - build: windows-2022-release
+          - build: windows-2022-release-amd
             os: [self-hosted, Windows, X64, "1ES.Pool=hld-win2022-amd"]
             hypervisor: none
```
Comment (ludfjig) [inline .github/workflows/Benchmarks.yml]: Optional: since we are refactoring some ci here, could we make the hypervisor different from `none` on windows? I'm thinking kvm/mshv/hyperv, where hyperv is for windows

File: .github/workflows/CreateRelease.yml [hunk 3]
```diff
@@ -155,29 +155,50 @@ jobs:
...
-      - name: Download benchmarks (Linux kvm)
+      - name: Download benchmarks (Linux kvm amd)
         uses: actions/download-artifact@v4
         with:
-          name: benchmarks_Linux_kvm
-          path: benchmarks_Linux_kvm
+          name: benchmarks_Linux_kvm_amd
+          path: benchmarks_Linux_kvm_amd
+
+      - name: Download benchmarks (Windows intel)
```
Comment (ludfjig) [inline .github/workflows/CreateRelease.yml]: Since there are a lot of this download-artifact action, maybe we could combine them into 1 using something like this? https://github.com/[actions/download-artifact](https://github.com/actions/download-artifact?tab=readme-ov-file#download-multiple-filtered-artifacts-to-the-same-directory)?tab=readme-ov-file#download-multiple-filtered-artifacts-to-the-same-directory

**Review by ludfjig — COMMENTED** [review]
lgtm with some minor questions

File: README.md (hunk 1 above)
Comment (danbugs) [reply README.md]: They are pretty minor README changes, so I figured it'd be alright. But ok, I'll move.

Comment (danbugs) [reply README.md]: Done: https://github.com/hyperlight-dev/hyperlight/pull/36

File: .github/workflows/Benchmarks.yml (hunk 2 above)
Comment (danbugs) [reply .github/workflows/Benchmarks.yml]: Changed the Windows name to "hyperv". Don't see a reason to change the Linux name, feel like that could lead to confusion.

File: .github/workflows/CreateRelease.yml (hunk 3 above)
Comment (danbugs) [reply .github/workflows/CreateRelease.yml]: Done ✅

**Review by simongdavies — APPROVED** [review]

**Review by ludfjig — APPROVED** [review]
LGTM
